File: backend/app/routes/feedback.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session
from ..database import get_db
from ..models import User, Feedback
from ..schemas import FeedbackCreate, FeedbackOut
from ..auth import get_current_user

router = APIRouter(prefix="/api/feedback", tags=["feedback"])


@router.post("", response_model=FeedbackOut, status_code=status.HTTP_201_CREATED)
def submit_feedback(
    body: FeedbackCreate,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    fb = Feedback(user_id=current_user.id, content=body.content)
    db.add(fb)
    db.commit()
    db.refresh(fb)
    return FeedbackOut.model_validate(fb)

File: frontend/src/api/feedback.ts
import apiClient from './client'

export interface FeedbackResponse {
  id: number
  user_id: number | null
  content: string
  created_at: string
}

export async function submitFeedback(content: string): Promise<FeedbackResponse> {
  const { data } = await apiClient.post<FeedbackResponse>('/feedback', { content })
  return data
}
